src/gui/ScintillaEditor.cc

```diff
@@ -1317,10 +1317,17 @@ void ScintillaEditor::setIndicator(const std::vector<IndicatorData>& indicatorDa
   }
 }
 
+void ScintillaEditor::resetToolTip(){
+  setToolTip("");
+}
+
 void ScintillaEditor::onIndicatorClicked(int line, int col, Qt::KeyboardModifiers state)
 {
-  if (!(state == Qt::ControlModifier || state == (Qt::ControlModifier | Qt::AltModifier))) return;
-
+  if (!(state == Qt::ControlModifier || state == (Qt::ControlModifier | Qt::AltModifier))){
+    setToolTip("to open links, use <b>CTRL + Click</b>");
+    QTimer::singleShot(5000,this,SLOT(resetToolTip()));
+    return;
+  }
```

Comment:
The behavior is indeed a bit strange regarding when the tooltip actually shows. After toying a bit with that, how about moving to the `indicatorRelease` instead of `indicatorClicked` event and go:
```
void ScintillaEditor::onIndicatorClicked(int line, int col, Qt::KeyboardModifiers state)
{
  qsci->SendScintilla(QsciScintilla::SCI_SETINDICATORCURRENT, hyperlinkIndicatorNumber);

  int pos = qsci->positionFromLineIndex(line, col);
  int val = qsci->SendScintilla(QsciScintilla::SCI_INDICATORVALUEAT, ScintillaEditor::hyperlinkIndicatorNumber, pos);

  // checking if indicator clicked is hyperlinkIndicator
  if (val >= hyperlinkIndicatorOffset && val <= hyperlinkIndicatorOffset + indicatorData.size()) {
    if (QGuiApplication::keyboardModifiers().testFlag(Qt::ControlModifier) || QGuiApplication::keyboardModifiers().testFlag(Qt::AltModifier)) {
      emit hyperlinkIndicatorClicked(val - hyperlinkIndicatorOffset);
    } else if (QGuiApplication::keyboardModifiers() == Qt::NoModifier) {
      QTimer::singleShot(0, this, [this] {
        QToolTip::showText(QCursor::pos(), "Use <b>CTRL + Click</b> to open the file", this, rect(), toolTipDuration());
      });
    }
  }
}
```
This almost retains the CTRL/ALT limitation of the original (just allows either combination of the two to open the file) and when clicking without any modifier it shows the tooltip directly.
The default time seems a bit long but as it hides when clicking somewhere else that might be ok instead of hard coding a time to show.
This also would not need the `resetToolTip()` method anymore.

Reply:
Oh, also, I'm not sure if that's currently relevant, but it seems better to have the tooltip logic inside the "ok, we checked it's actually a hyperlink indicator" as the callback triggers for any indicators set in the text.

Reply:
I will look into it. The suggestions sound good.

Currently, there are no other uses for indicators.

Reply:
ah, clever. The `Qt::NoModifier`check  is to ignore Shift+Click, which is used by the editor.

Regarding [`toolTipDuration()` ](https://doc.qt.io/qt-5/qwidget.html#toolTipDuration-prop)

> Specifies how long time the tooltip will be displayed, in milliseconds.
> If the value is -1 (default) the duration is calculated depending on the length of the tooltip.

Sound fine.

> moving to the `indicatorRelease` instead of `indicatorClicked` event

so that the tooltip is not shown on click+drag?

> The default time seems a bit long but as it hides when clicking somewhere else that might be ok instead of hard coding a time to show.

Strange. I am under `Ubuntu 21.10` which is using `Qt version 5.15.2 `.
There, using your code, the tool-tip disappears immediately when releasing the mouse button.

Or your describe what happens after _moving to the `indicatorRelease` instead of `indicatorClicked`_?

Reply:
Yes, the *disappears immediately* happens with the `indicatorClicked` event. I have not found a clear documentation describing that behavior, but I suppose it's the release event doing the cancel action. And not even the `QTimer` trick works. But I think the release event does make sense anyway.
For some reason I can't get the keyboard modifiers in the `indicatorRelease` event, it's always `0`. But the global keyboard state should be fine.